pennylane_lightning/experimental/lightning_qubit_2.py

```diff
@@ -0,0 +1,171 @@
+
+            This device:
+
+            * Supports any qubit operations that provide a matrix
+            * Currently does not support finite shots
+            * Currently does not intrinsically support parameter broadcasting
+
+            """
+            is_single_circuit = False
+            if isinstance(circuits, QuantumScript):
+                circuits = [circuits]
+                is_single_circuit = True
+
+            batch, post_processing_fn = preprocess(circuits, execution_config=execution_config)
+
+            if is_single_circuit:
+
+                def convert_batch_to_single_output(results):
+                    """Unwraps a dimension so that executing the batch of circuits looks like executing a single circuit."""
+                    return post_processing_fn(results)[0]
+
+                return batch, convert_batch_to_single_output
+
+            return batch, post_processing_fn
+
+        def execute(
+            self,
+            circuits: QuantumTape_or_Batch,
+            execution_config: ExecutionConfig = DeviceExecutionConfig,
+        ):
+            """Execute a circuit or a batch of circuits and turn it into results.
+
+            Args:
+                circuits (Union[QuantumTape, Sequence[QuantumTape]]): the quantum circuits to be executed
+                execution_config (Union[ExecutionConfig, Sequence[ExecutionConfig]]): a data structure with additional information required for execution
+
+            Returns:
+                TensorLike, tuple[TensorLike], tuple[tuple[TensorLike]]: A numeric result of the computation.
+            """
+            is_single_circuit = False
+            if isinstance(circuits, QuantumScript):
+                is_single_circuit = True
+                circuits = [circuits]
+
+            if self.tracker.active:
+                self.tracker.update(batches=1, executions=len(circuits))
+                self.tracker.record()
+
+            results = tuple(simulate(c) for c in circuits)
+            return results[0] if is_single_circuit else results
+
+        def supports_derivatives(
+            self,
+            execution_config: ExecutionConfig,
+            circuit: Optional[QuantumTape] = None,
+        ) -> bool:
+            """Check whether or not derivatives are available for a given configuration and circuit.
+
+            ``LightningQubit2`` supports adjoint differentiation method.
+
+            Args:
+                execution_config (ExecutionConfig): The configuration of the desired derivative calculation
+                circuit (QuantumTape): An optional circuit to check derivatives support for.
+
+            Returns:
+                Bool: Whether or not a derivative can be calculated provided the given information
+            """
+            if (
+                execution_config.gradient_method != "adjoint"
+                or execution_config.derivative_order != 1
+            ):
+                return False
```

Comment:
I think all good for now. We can probably update this later to allow custom derivative operations to be registered and validated, with some extra list/dictionary storage on the device as needed.